wger.py

```diff
@@ -127,186 +161,188 @@ def create_exercise(self):
+                      "name": workout_name,
+                      "description": ""}
         self.post_info('workout/',workout_data)
-        for i in self.get_info('workout/')['results']:
+        for i in self.get_info('workout/').get('results'):
             if i['name']==workout_name:
-                id_workout=i['id']
+                id_workout=i.get('id')
 
-        while(counter<nb_days):
-            data_day={"id":"","training":id_workout ,"description": f"Day {counter+1}","day": [counter+1]}
+        while(counter<no_days):
+            data_day={"id":"",
+                      "training":id_workout ,
+                      "description": f"Day {counter+1}",
+                      "day": [counter+1]}
             self.post_info('day/',data_day)
-            self.select_exercises(workout_name,data_day['description'])
+            self.select_exercises(workout_name,data_day['description'],6,2)
             counter+=1
-        return f"Workout {workout_name} created successfully!"
-
+        return any([i['id']==id_workout for i in user1.get_info('workout/').get('results')])
 
 
-    def schedule(self,schedule_name,start_date):
+    def schedule(self,schedule_name,start_date,is_loop,is_active):
         """This function creates a new schedule.The schedule will be active and will have a given start date"""
-        data={"id":"","name": schedule_name,"start_date": start_date,"is_active": True,"is_loop": True}
+        data={"id":"",
+              "name": schedule_name,
+              "start_date": start_date,
+              "is_active": is_active,
+              "is_loop": is_loop}
         response=self.post_info('schedule/',data)
         return response
 
     def workout_in_schedule(self,schedule_name,workout_name,duration):
         """This function adds an existing workout to a schedule"""
         schedules=self.get_info('schedule/')['results']
         workouts=self.get_info('workout/')['results']
-        id_schedule=""
-        id_workout=""
-        for i in schedules:
-            if i['name']==schedule_name:
-                id_schedule=i['id']
-        if id_schedule=="":
-            return "Schedule not found"
-
-        for i in workouts:
-            if i['name']==workout_name:
-                id_workout=i['id']
-        if id_workout=="":
-            return "Workout not found"
-
-        data={"id":"","schedule": id_schedule,"workout":id_workout,"duration": duration}
+        id_schedule=self.match(schedules,schedule_name,'name','id')
+        id_workout=self.match(workouts, workout_name, 'name', 'id')
+        data={"id":"",
+              "schedule": id_schedule,
+              "workout":id_workout,
+              "duration": duration}
         response=self.post_info('schedulestep/',data)
         return response
 
-    def delete_exercise(self,workout,day,exercise):
+
+    def delete_exercise(self,workout,day=None,exercise=None):
         """The function delete an exercise from a specific training day"""
```

Comment:
I think you misunderstood the logic needed in this method:
![image](https://user-images.githubusercontent.com/60219256/119654113-c5225000-be30-11eb-8f37-2c748887d380.png)